components/ui-api-layer/internal/domain/k8s/resourcequota_resolver.go

```diff
@@ -30,3 +71,153 @@ func (r *resourceQuotaResolver) ResourceQuotasQuery(ctx context.Context, environ
 
 	return r.converter.ToGQLs(items), nil
 }
+
+func (r *resourceQuotaResolver) ResourceQuotaStatus(ctx context.Context, environment string) (gqlschema.ResourceQuotaStatus, error) {
+	const (
+		rsKind     = "ReplicaSet"
+		ssKind     = "StatefulSet"
+		deployKind = "Deployment"
+	)
+
+	resourcesToCheck := []v1.ResourceName{
+		v1.ResourceRequestsMemory,
+		v1.ResourceLimitsMemory,
+		v1.ResourceRequestsCPU,
+		v1.ResourceLimitsCPU,
+		v1.ResourcePods,
+	}
+
+	resourceQuotas, err := r.rqLister.ListResourceQuotas(environment)
```

Comment:
there are 3 checks - please add a comment how does every check is working (maybe you can paste a proper statement from the proposal), it is necessary for statefulsets and replicasets which are more tricky.

Reply:
Added the comment above each check.